js/civicrm_stripe.js

```diff
@@ -151,14 +151,12 @@
       // Hide payment if total is 0 and no more participants.
       if ($('#priceset').length) {
         additionalParticipants = cj("#additional_participants").val();
-        // The currentTotal is already being calculated in Form/Contribution/Main.tpl.
-        if(typeof currentTotal !== 'undefined') {
-          if (currentTotal == 0 && !additionalParticipants) {
-            // This is also hit when "Going back", but we already have stripe_token.
-            debugging('ozlkf');
-            // This should not happen on Confirm Contribution, but seems to on 4.6 for some reason.
-            //return true;
-          }
+        currentTotal = $('#pricevalue').data('raw-total');
+        if (currentTotal == 0 && !additionalParticipants) {
+          // This is also hit when "Going back", but we already have stripe_token.
+          debugging('ozlkf');
+          // This should not happen on Confirm Contribution, but seems to on 4.6 for some reason.
+          //return true;
         }
```

Comment:
The comments tell the story.....it was breaking 4.6 transactions. 

`//This should not happen on Confirm Contribution, but seems to on 4.6 for some reason.`

Do you mind testing this PR with that line uncommented on 4.6?